File: javascript/grade-school/grade-school.js
var School = function() {
  "use strict";

  var roster = {};

  function getRoster() {
    return roster;
  }

  function setRoster(name, grade) {
    if (roster[grade]) {
      roster[grade].push(name);
      roster[grade].sort();
    } else {
      roster[grade] = [name];
    }
  }

  function grade(grade) {
    if (roster[grade]) {
      return roster[grade];
    } else {
      return [];
    }
  }

  return {
    roster: getRoster,
    add: setRoster,
    grade: grade
  };
};

module.exports = School;

File: python/grade-school/grade_school.py
from collections import defaultdict

class School:
    def __init__(self):
        self.db = defaultdict(list)

    def add_student(self, name, grade):
        self.db[grade].append(name)
        self.db[grade] = sorted(self.db[grade])

    def roster(self):
        pass

    def grade(self, grade_number):
        return self.db[grade_number]
